jetty-ee10/jetty-ee10-proxy/src/main/java/org/eclipse/jetty/ee10/proxy/AsyncMiddleManServlet.java

```diff
@@ -594,6 +595,7 @@ public void failed(Throwable failure)
 
     protected class ProxyWriter implements WriteListener
     {
+        private final AutoLock lock = new AutoLock();
         private final Queue<Chunk> chunks = new ArrayDeque<>();
```

Comment:
`ProxyResponseListener.buffers` also looks like a collection which needs to be threadsafe.